conmon/conmon.c

```diff
@@ -1329,6 +1331,25 @@ int main(int argc, char *argv[])
 		if (dup2(slavefd_stderr, STDERR_FILENO) < 0)
 			pexit("Failed to dup over stderr");
 
+		/* If LISTEN_PID env is set, we need to set the LISTEN_PID
+		   it to the new child process */
+		char *listenpid = getenv("LISTEN_PID");
+		if (listenpid != NULL) {
+			char *endptr=0;
+			errno=0;
+			int lpid = strtol(listenpid, 0, 10);
+			if (errno != 0 || *endptr != '\0')
```

Comment:
If you're going to use `endptr` as part of your check, you probably want to pass it to `strtol`.  There's an example showing this in [the Linux man page][1].  But since we have stricter limits on valid PIDs, I think we can skip `endptr` and use something like:

```C
errno = 0;
long lpid = strtol(listenpid, NULL, 10);
if (errno != 0 || lpid <= 0)
  pexit("Invalid LISTEN_PID %s: %s", listenpid, strerror(errno));
```

[1]: http://man7.org/linux/man-pages/man3/strtol.3.html